packages/react/select/src/Select.tsx

```diff
@@ -334,7 +334,7 @@ const SelectValue = React.forwardRef<SelectValueElement, SelectValueProps>(
         // through the item they came from
         style={{ pointerEvents: 'none' }}
       >
-        {context.value === undefined && placeholder !== undefined ? placeholder : children}
+        {shouldShowPlaceholder(context.value) ? <>{placeholder}</> : children}
```

Comment:
The key thing here is the fragment, which enables React to find its feet and reconcile properly when we reset the state to "". This is because when a value is set, we portal into that node, and without that React seems to lose it and isn't able to render the placeholder again.